### PR TITLE
Unpin numpy as a host dep

### PR DESCRIPTION
**Pull request by MilesCranmer**
Continuation of #85

cc @h-vetinari @ngam @mkitti

### CODE REVIEW

**Comment by MilesCranmer** [comment]
Damn, still broken: https://github.com/MilesCranmer/PySR/actions/runs/5192076497

**Comment by mkitti** [comment]
What is the load error to begin with? Set `LD_DEBUG=libs` for Linux.

**Comment by MilesCranmer** [comment]
Not sure. I think if I fix the `libdstdcxx-ng` to an earlier version the error goes away (IIRC).

Here's a run with `LD_DEBUG=libs` (just started)

**Comment by h-vetinari** [comment]
> Damn, still broken: https://github.com/MilesCranmer/PySR/actions/runs/5192076497

You have a completely separate [issue](https://github.com/conda-forge/scipy-feedstock/issues/238) there; it _looks_ like it's scipy, but it's not. I've [tested](https://github.com/conda-forge/scipy-feedstock/pull/240) it. It might however be influenced if something similar to the numpy situation is happening for scipy (e.g. reinstalling it through julia).

**Comment by h-vetinari** [comment]
Rather than a two-step setup, why not include `python -m pysr.test main` (with whatever preparations are necessary) here?

**Comment by h-vetinari** [comment]
> If I want to constrain `libstdcxx-ng'<13'`, I guess I should add that to `host:`?

You would have to add it under `run_constrained:`. But it shouldn't be necessary, so I'd be interested how it fails without that pin.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,17 +10,17 @@ source:
...
     - pip
     - pyjulia >=0.6.0
     - julia
-    # temporarily rebuilding with newest numpy because PyCall.jl ends up
-    # reinstalling it and we want to avoid spamming ClobberWarnings
-    # TODO: remove this as soon as numpy does not get spuriously installed here anymore
+    - numpy
+    # Temporary fix for PyCall.jl mistakenly re-installing numpy.
+    # TODO: Remove this.
     - numpy >=1.24.3
```
Comment (h-vetinari) [inline recipe/meta.yaml:24]: This should be removed now

**Comment by mkitti** [comment]
> I also thought this post was quite interesting about how the whole gig works: https://discourse.julialang.org/t/the-present-and-the-future-of-package-registration/99890

What does that have to do with this?

It's basically analgous to a conda channel. We can create our own registry:
https://github.com/GunnarFarneback/LocalRegistry.jl

I just don't want to run it.

**Comment by MilesCranmer** [comment]
Superseded by https://github.com/conda-forge/pysr-feedstock/pull/90

**Pull request closed without merging**